app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/SearchInputView.kt

```diff
@@ -97,7 +121,11 @@ class SearchInputView(val state: RHMIState,
 			}
 
 			override fun onOk() {
-				browseController.showSearchResults(deferredSearchResults, inputComponent.getResultAction()?.asHMIAction())
+				if(!isSpotify && deferredSearchResults.isCompleted && deferredSearchResults.getCompleted()?.isEmpty() == true) {
+					browseController.playFromSearch(inputComponent.getResultAction()?.asHMIAction(), this.input)
+				} else {
```

Comment:
I like this good start! I think the `showSearchResults` shouldn't be the final `else` case, because relatively few music controllers will provide search results (though the first branch should catch unsearchable apps with a quickly-resolved empty list). I think `showSearchResults` should only be called when `!deferredSearchResults.isCompleted || deferredSearchResults.getCompleted()?.isNotEmpty` (to be used if it is actually loading or has already loaded some results), and the else branch should just not do anything, except perhaps raise an RHMIActionAbort to ignore any previous HMIAction target state.

Reply:
Ah got it, changed that else to an else if with the conditions that it is not completed or if it is then it is not empty then showSearchResults will be called and omitted the else branch. I don't think we'll need to throw a RHMIActionAbort since the target state will not be set at all as it is only set through the playFromSearch() or showSearchResults method so it should remain unchanged in this else branch

Reply:
Excellent! It should be a rare situation when it would fall through to the else branch anyways.
Keep in mind that the HMIEvent's Target State isn't cleared during the app's lifetime and will remember whichever last state was last set. While it will start out as not going anywhere, any previous playFromSearch or showSearchResults will set this state and then future invocations of this branch won't remain on this InputState as intended. I think this case will be very rare, though, because the first two branches should cover most situations.